File: src/test/java/com/vassant/kata/domain/HistoryTest.java
package com.vassant.kata.domain;

import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;
import java.util.List;

import static java.util.Arrays.asList;
import static org.assertj.core.api.Assertions.assertThat;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
class HistoryTest {

    @Test
    void should_sort_operations_by_date() {
        LocalDateTime TODAY = LocalDateTime.now();
        LocalDateTime TOMORROW = TODAY.plusDays(1);
        LocalDateTime NEXT_WEEK = TODAY.plusWeeks(1);
        List<Operation> operation = asList(aOperationAt(TOMORROW), aOperationAt(TODAY), aOperationAt(NEXT_WEEK));

        History history = new History(operation);

        assertThat(history.getOperation()).extracting("date").containsExactly(TODAY, TOMORROW, NEXT_WEEK);
    }

    private Operation aOperationAt(LocalDateTime TOMORROW) {
        return new Operation(TOMORROW, OperationType.DEPOSIT, Amount.of(100), Balance.of(0));
    }
}

File: src/test/kotlin/com/vassant/kata/domain/HistoryTest.kt
package com.vassant.kata.domain

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.groups.Tuple
import org.junit.jupiter.api.Test
import java.time.LocalDateTime
import java.util.*

class HistoryTest {

    @Test
    fun `should sort operations by date`() {
        val TODAY = LocalDateTime.now()
        val TOMORROW = TODAY.plusDays(1)
        val NEXT_WEEK = TODAY.plusWeeks(1)
        val operation = Arrays.asList(aOperationAt(TOMORROW), aOperationAt(TODAY), aOperationAt(NEXT_WEEK))

        val history = History(operation)

        assertThat(history.lines).extracting("operation.date", "balance")
                .containsExactly(
                        Tuple(TODAY, Balance(100)),
                        Tuple(TOMORROW, Balance(200)),
                        Tuple(NEXT_WEEK, Balance(300))
                )
    }

    private fun aOperationAt(dateOperation: LocalDateTime): Operation {
        return Deposit(dateOperation, Amount(100), Balance(0))
    }
}